sycl/source/detail/persistent_device_code_cache.cpp

```diff
@@ -78,10 +78,13 @@ void PersistentDeviceCodeCache::putItemToDisc(
     const SerializedObj &SpecConsts, const std::string &BuildOptionsString,
     const RT::PiProgram &NativePrg) {
 
+  if (!isImageCached(Img))
+    return;
+
   std::string DirName =
       getCacheItemPath(Device, Img, SpecConsts, BuildOptionsString);
```

Comment:
FYI. @vladimirlaz, https://github.com/intel/llvm/pull/5240/commits/cd9818b861d36ae389d640b93e1c193c5235b094 avoids calling `getCacheItemPath` because "it's expensive" (here I mean it emits a few calls to the backed) and all this work is wasted if cache is disabled.
I noticed that while I was coding unit-test and had to mock a lot of back-end/plug-in APIs.